kratos/utilities/function_parser_utility.h

```diff
@@ -0,0 +1,261 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborator:    Vicente Mataix Ferrandiz
+//
+
+#if !defined(KRATOS_FUNCTION_PARSER_UTILITY_H_INCLUDED)
+#define  KRATOS_FUNCTION_PARSER_UTILITY_H_INCLUDED
+
+// System includes
+#include <unordered_map>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+
+// Forward declaration to avoid the include of tinyexpr in a header file
+struct te_expr;
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class GenericFunctionUtility
+ * @ingroup KratosCore
+ * @brief This function allows to call a function method of type f(x, y, z, t) implemented in python style.
+ * @details The functions can be constructed by providing a python-defined method of the type
+ *
+ *
+ * The object is then insantiated as
+ * aux_function = GenericFunctionUtility(aux_object_cpp_callback(self.function_string))
+ *
+ * Optionally one can specify a rotation matrix and an origin so that the function can be defined in a rotated system of coordinates
+ * @author Riccardo Rossi
+ * @author Vicente Mataix Ferrandiz
+ */
+class KRATOS_API(KRATOS_CORE) GenericFunctionUtility
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    /// The index type definition
+    typedef std::size_t IndexType;
+
+    /// Counted pointer of GenericFunctionUtility
+    KRATOS_CLASS_POINTER_DEFINITION(GenericFunctionUtility);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     * @param rFunctionBody The string defining the function
+     * @param LocalSystem The parameters defining the local system
+     */
+    GenericFunctionUtility(
+        const std::string& rFunctionBody,
+        Parameters LocalSystem = Parameters{}
+        );
+
+    ///Copy constructor
+    GenericFunctionUtility(GenericFunctionUtility const& rOther);
+
+    /// Destructor.
+    ~GenericFunctionUtility();
```

Comment:
here you would have to destroy the things you allocate in row pointers

Reply:
Is in the cpp